sphinx/ext/autodoc.py

```diff
@@ -514,18 +515,22 @@ def process_doc(self, docstrings):
             for line in docstringlines:
                 yield line
 
-    def add_content(self, more_content, no_docstring=False):
-        """Add content from docstrings, attribute documentation and user."""
-        # set sourcename and add content from attribute documentation
+    def get_sourcename(self):
         if self.analyzer:
             # prevent encoding errors when the file name is non-ASCII
-            if not isinstance(self.analyzer.srcname, text_type):
-                filename = text_type(self.analyzer.srcname,
-                                     sys.getfilesystemencoding(), 'replace')
+            if not isinstance(self.analyzer.srcname, unicode):
+                filename = unicode(self.analyzer.srcname,
+                                   sys.getfilesystemencoding(), 'replace')
             else:
                 filename = self.analyzer.srcname
-            sourcename = u'%s:docstring of %s' % (filename, self.fullname)
+            return u'%s:<docstring of %s>' % (filename, self.fullname)
```

Comment:
This is the cause of the test failures: the angle brackets are added compared to the old return value.

I think it's better readable with angle brackets, but:

* test should be adapted
* angle brackets should also be added in the next line for consistency